app/security/Roles.scala

```diff
@@ -347,6 +347,8 @@ object RoleUtils {
 
   def isSdipFaststream(implicit user: CachedData): Boolean = user.application exists (_.applicationRoute == ApplicationRoute.SdipFaststream)
 
+  def isSdipFaststreamFailed(implicit user: CachedData): Boolean = user.application.fold(false)(_.progress.sdipFSFailed)
```

Comment:
Not sure a getOrElse would work as we're accessing fields in nested classes.  Happy to merge with the fold though.